src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,160 @@
+
+        // The list includes dates to be monitored in a 7-day window starting from the query date.
+        List<String> newDates = List.of(QUERY_DATE /* Thursday */, "2020-08-15", "2020-08-16", "2020-08-17", "2020-08-18");
+        List<String> checkedDates = ItineraryUtils.getDatesToCheckItineraryExistence(trip);
+        Assertions.assertEquals(newDates, checkedDates);
+    }
+
+    @Test
+    public void testAddIgnoreRealtimeParam() throws URISyntaxException {
+        String queryWithRealtimeParam = BASE_QUERY + "&" + IGNORE_REALTIME_UPDATES_PARAM + "=false";
+        List<String> queries = List.of(BASE_QUERY, queryWithRealtimeParam);
+
+        for (String query : queries) {
+            String queryExcludingRealtime = ItineraryUtils.excludeRealtime(query);
+            Map<String, String> params = ItineraryUtils.getQueryParams(queryExcludingRealtime);
+            Assertions.assertEquals("true", params.get(IGNORE_REALTIME_UPDATES_PARAM));
+        }
+    }
+
+    @Test
+    public void testUpdateTripWithVerifiedItinerary() throws IOException, URISyntaxException {
+        String query = BASE_QUERY + "&ui_activeItinerary=1";
+
+        String mockResponse = FileUtils.getFileContents(
+            TEST_RESOURCE_PATH + "persistence/planResponse.json"
+        );
+        OtpDispatcherResponse otpDispatcherResponse = new OtpDispatcherResponse(mockResponse);
+        List<Itinerary> itineraries = otpDispatcherResponse.getResponse().plan.itineraries;
+
+        MonitoredTrip trip = new MonitoredTrip();
+        trip.id = "testUpdateTripWithVerifiedItinerary";
+        trip.queryParams = query;
+        // Write some itinerary that is not the one we want.
+        trip.itinerary = itineraries.get(0);
+
+        ItineraryUtils.updateTripWithVerifiedItinerary(trip, itineraries);
+
+        Assertions.assertEquals(itineraries.get(1), trip.itinerary);
+    }
+
+    private void testItineraryDepartsSameDay(String time, boolean expected, Long... startTimes) {
+        // Trip is in US Eastern timezone per the place set in makeBarebonesTrip().
+        MonitoredTrip trip = makeBarebonesTrip();
+        ZoneId zoneId = trip.tripZoneId();
+
+        // startTimes are in US Eastern timezone.
+        for (Long startTime : startTimes) {
+            Itinerary itinerary = new Itinerary();
+            Instant instant = Instant.ofEpochMilli(startTime);
+            itinerary.startTime = Date.from(instant);
+            Assertions.assertEquals(
+                expected,
+                ItineraryUtils.itineraryDepartsSameDay(itinerary, QUERY_DATE, time, zoneId),
+                String.format(
+                    "%s %s be considered same day as %s %s",
+                    ZonedDateTime.ofInstant(instant, zoneId),
+                    expected ? "should" : "should not",
+                    QUERY_DATE,
+                    time
+                )
+            );
+        }
+    }
+
+    @Test
+    public void testItineraryDepartsSameDay() {
```

Comment:
I like these tests, but think we need some more for trips that "arrive by".

Reply:
I added the arrive-by tests in https://github.com/ibi-group/otp-middleware/pull/77/commits/0aba7da8c43c1e2b26da517ee70d6f821b9a933b